source/Calamari.AzureServiceFabric/Integration/AzureServiceFabricPowerShellContext.cs

```diff
@@ -90,6 +102,45 @@ public CommandResult ExecuteScript(Script script,
             }
         }
 
+        async Task<string> GetAzureADAccessToken(string serverCertThumbprint, string connectionEndpoint, string aadUserCredentialUsername, string aadUserCredentialPassword)
```

Comment:
I _feel_ like this could be pulled out and made common between the health-check and here - maybe even pulling out the whole switch-statement around auth-type.

Reply:
or maybe not - you are doing slightly different things